compare.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/python3
+
+import koji
+
+import argparse
+import datetime
+import jinja2
+import logging
+import os
+import re
+import requests
+import rpm
+
+
+class BuildSource:
+
+    def __init__(self, source_id=None, infra=None, tag=None, make_cache=True):
+        """Setup a source of the builds
+
+        :infra: Koji or Brew session,
+        :tag: Koji or Brew tag
+        """
+
+        if source_id:
+            infra, tag = self._configure_source(source_id)
+
+        self.infra = infra
+        self.tag = tag
+        self.cache = {}
+
+        if make_cache:
+            self.make_cache()
+
+    def __str__(self):
+        return f'{self.tag}'
+
+    def _configure_source(self, source_id):
+        if source_id == "rawhide":
+            infra = koji.ClientSession('https://koji.fedoraproject.org/kojihub')
+            tag = infra.getFullInheritance('rawhide')[0]['name']
+        if source_id == "eln":
+            infra = koji.ClientSession('https://koji.fedoraproject.org/kojihub')
+            tag = "eln"
+        if source_id == "rhel":
+            # FIXME
+            infra = koji.ClientSession('https://brew')
+            tag = "rhel-9.0.0-alpha"
```

Comment:
I'd also added a condition here where `source_id ` doesn't equal all these cases.
It will help to debug if something goes wrong.